src/plaf/backends/halo2.rs

```diff
@@ -22,6 +22,44 @@ pub struct PlafH2Circuit {
     pub wit: Witness,
 }
 
+impl PlafH2Circuit {
+    pub fn instance<F: PrimeField<Repr = [u8; 32]>>(&self) -> Vec<Vec<F>> {
+        if !self.plaf.columns.public.is_empty() {
+            let mut instance_with_offsets = Vec::new();
+            for copy in &self.plaf.copys {
+                let (left_col, right_col) = &copy.columns;
+
+                let (witness_col, offsets) = match (left_col.kind, right_col.kind) {
+                    (ColumnKind::Witness, ColumnKind::Public) => (left_col, copy.offsets.clone()),
+                    (ColumnKind::Public, ColumnKind::Witness) => (
+                        right_col,
+                        copy.offsets.iter().map(|(l, r)| (*r, *l)).collect(),
+                    ),
+                    _ => continue,
```

Comment:
In order to prevent silently skipping https://github.com/Dhole/polyexen/pull/3#issuecomment-1732344916 can you add an entry to the match like this?
```rust
(ColumnKind::Public, _) | (_, ColumnKind::Public) => unimplemented!("..."),
```